crates/driver/src/settlement_proposal.rs

```diff
@@ -47,13 +47,13 @@ impl TradedOrder {
     /// User orders are allowed to get surplus and therefore return the clearing price of the
     /// buy_token whereas liquidity orders must not get surplus so they return their limit price.
     fn buy_token_price(&self, clearing_prices: &HashMap<H160, U256>) -> Option<U256> {
-        match self.order.metadata.is_liquidity_order {
-            // liquidity orders have to be settled at their limit price
-            true => clearing_prices
+        if self.order.metadata.class == OrderClass::Ordinary {
+            clearing_prices.get(&self.order.data.buy_token).cloned()
+        } else {
+            clearing_prices
                 .get(&self.order.data.sell_token)?
                 .checked_mul(self.order.data.sell_amount)?
-                .checked_div(self.order.data.buy_amount),
-            false => clearing_prices.get(&self.order.data.buy_token).cloned(),
+                .checked_div(self.order.data.buy_amount)
```

Comment:
nit: My personal preference is to use `match` in these cases when all match cases are handled (for clarity), but feel free to ignore.

Reply:
Since this is an enum I also feel that match that is more natural than if-else with equality comparison of enum values in the condition. Can still use `_ =>` like `else`. Not a big deal, just feels better to me.

Reply:
https://github.com/cowprotocol/services/pull/670/commits/f068f1187d88dc9c3c34622f4300088ae9947213

@vkgnosis Personally I tend to often avoid using `_` in match unless I'm sure I really mean it. The reason when you don't use `_` and you add a new variant to the enum, you get a compiler error which forces you to revisit the code and make sure that it's still correct. Whereas `_` implicitly matches the new variant without warning you.